### DISCUSSION TITLE
在下一个大版本对“CQ码”更名

### DISCUSSION THREAD

**Discussion by Ink-33** [howmanybots/onebot#77] [想法 / 建议]
众所周知CQ码这一名词已经成为了历史，继续沿用这一名词是否并不妥当？
个人认为在下一个大版本中，可以将`CQ码`更名为`OB码`；`CQ:at`等功能块更名为`OB:at`，亦或者将`CQ:`功能块的写法列为not recommend
如果有其他意见可以继续探讨，以上。

**Comment by Ink-33** [comment]
@Pai2Chen 我觉得将CQ码列入不推荐写法，新协议推荐使用OB码兼容多平台。到底要不要兼容CQ码端写法还是看机器人框架开发者

**Comment by zhzhongshi** [comment]
保留CQ码（兼容用），建议使用OB码，

**Comment by Tamce** [comment]
现在基于CQ的框架和插件太多了...一时半会儿确实不太好改

**Comment by L1yp** [comment]
> 现在基于CQ的框架和插件太多了...一时半会儿确实不太好改

发送隐式支持CQ码, 接收去掉?

**Comment by DZ-IO** [comment]
我觉得酷q老API兼容的部分继续使用CQ码，新加入的功能使用OB码就是个不错的选择

不应该去除CQ码，也是对酷q的一种纪念对不

**Comment by mzdluo123** [comment]
我建议删除CQ码，CQ码作为cq时期因编程语言功能的限制而引入的折中方案在现在已经不必要。
现在使用json序列化消息的方式已经能够胜任大部分工作情况，同时使用json方式序列化消息也降低了onebot标准实现者的工作量

**Comment by Him188** [comment]
我建议在兼容模块（或者一个独立的插件）来兼容 CQ 码，而不是修改 OneBot 标准来适合 CQ 码。目前各大机器人核心已经有了兼容 CQ 的实现（插件），故 OneBot 新版本删除 CQ 码不会有影响。OneBot 标准可以往现代化高效率 API 方向进行。

**Comment by heerheer** [comment]
作为从CQ时期来的我觉得保留CQ码还能起到纪念作用XD

**Comment by MasterYuan418** [comment]
建议新推出的都使用OB码，但任然保持对CQ码的支持

**Reply by crazywhalecc** [reply]
新标准经商讨应该不会原生支持 CQ 码或者 OB 码，但是 LibOneBot 或 OneBot 实现者可以通过扩展字段的方式兼容 CQ 码，如使用 `old_format_message` 字段。
